tests/ga/test_extension.py

```diff
@@ -2322,324 +2324,320 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
-        }
+        StatusBlobs and command invocations for each actor can be checked with
+        {emulator}.status_blobs and {emulator}.actions[{command_name}] respectively.
 
-        if handler:
-            handler_json.update(handler)
+        Note that this method assumes the first extension's install command should
+        succeed. Don't use this method if your test is attempting to emulate a fresh install
+        (i.e. not an upgrade) with a failing install.
+        """
 
-        ext_handler_properties = ExtHandlerProperties()
-        ext_handler_properties.version = version
-        ext_handler = ExtHandler(name=name)
-        ext_handler.properties = ext_handler_properties
-        ext_handler_i = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
-        ext_handler_i.load_manifest = MagicMock(return_value=HandlerManifest({'handlerManifest': handler_json}))
-        fileutil.mkdir(ext_handler_i.get_base_dir())
-        return ext_handler_i
+        with mock_wire_protocol(DATA_FILE, http_put_handler=generate_put_handler(first_ext, upgraded_ext)) as protocol:
 
-    def test_disable_failed_env_variable_should_be_set_for_update_cmd_when_continue_on_update_failure_is_true(
-            self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=True)
+            exthandlers_handler = get_exthandlers_handler(protocol)
 
-        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command",
-                          side_effect=ExtensionError('disable Failed')) as patch_start_cmd:
-            with self.assertRaises(ExtensionError):
-                ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+            with enable_invocations(first_ext, upgraded_ext) as invocation_record:
+                exthandlers_handler.run()
 
-            args, kwargs = patch_start_cmd.call_args
+                invocation_record.compare(
+                    (first_ext, ExtensionCommandNames.INSTALL),
 
-            self.assertTrue('-update' in kwargs['command'] and ExtCommandEnvVariable.DisableReturnCode in kwargs['env'],
-                            "The update command should have Disable Failed in env variable")
+                    # Note that if installCommand is supposed to fail, this will erroneously raise.
+                    (first_ext, ExtensionCommandNames.ENABLE)
+                )
 
-    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(
-            self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        old_handler_i.set_handler_state(ExtHandlerState.Enabled)
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=True)
+            protocol.mock_wire_data.set_extensions_config_version("1.1.0")
```

Comment:
Shouldn't you be picking up this version from `upgraded_ext.version` instead?

Reply:
You're right. Good catch.